narwhal/node/src/benchmark_client.rs

```diff
@@ -51,8 +56,60 @@ struct App {
     client_metric_host: String,
     #[clap(long, default_value = "8081", global = true)]
     client_metric_port: u16,
+    // Local or remote client operating mode.
+    #[clap(long, default_value = "remote", value_parser)]
+    operating_mode: OperatingMode,
+}
+
+#[derive(Clone)]
+pub enum OperatingMode {
+    // Submit transactions via local channel
+    Local,
+    // Submit transactions via grpc
+    Remote,
+}
+
+impl std::str::FromStr for OperatingMode {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "local" => Ok(OperatingMode::Local),
+            "remote" => Ok(OperatingMode::Remote),
+            _ => Err("must be 'local' or 'remote'".to_string()),
+        }
+    }
+}
+
+#[async_trait::async_trait]
+pub trait SubmitToConsensus: Sync + Send + 'static {
```

Comment:
How is this trait definition and implementation used?

Reply:
consensus_adapter.rs impl
```
async fn submit_to_consensus(
        &self,
        transaction: &ConsensusTransaction,
        _epoch_store: &Arc<AuthorityPerEpochStore>,
    ) -> SuiResult {}
```

It is just a custom trait so I can call `submit_to_consensus` from LazyNarwhalClient without having to pass in epoch store. But now that you ask I probably could have just created a method that took in LazyNarwhalClient as an arg. Will change it